update PR template to consider accessibility when reviewing (#14277)
rancher/dashboard: .github/pull_request_template.md

- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

> - [ ] The PR has been reviewed in terms of Accessibility (test keyboard navigation, use screen reader, check ARIA roles and attributes)

Reviewer: ```suggestion
- [ ] The PR has been reviewed in terms of Accessibility, or there are no HTML changes
```

These should be brief and always able to be checked.

Author: I am going to update/create a Confluence page just to address @rak-phillip comment. Overall internal documentation on a11y is a bit overdue.

I understand your suggested change but I don't feel that the path forward is to create "external" docs. What I have between parentheses is the short version of what people should look out for, if applicable ofc. 

If I can't write it because everything might not be applicable, then where does developer awareness and "wiseness" enters in this equation? 

Developers should already be aware of what does accessibility mean. I don't think we need to explain not to check ARIA roles and attributes if no HTML has changed. We need to draw the line somewhere.

I still think is beneficial the items inside the parentheses rather than removing it because it's a quick check guide, but I definitely don't agree with creating external documentation, especially for this.

CC @nwmac

Author: Created two pages in Confluence:
- General developer guide
- What to look out for when reviewing PRs

DM me to get those links, if needed

@rak-phillip @richard-cox @nwmac

Reviewer 2: I would actually tend to agree that the shorter string is better, I would lose the no HTML changes and just say `Thee PR has been reviewed in terms of Accessibility`. As you say, "Developers should already be aware of what does accessibility mean", so I don't think we need to add any detail other than what they are adding should be accessibility compliant.

Author: @nwmac updated
